internal/driver/drv_uart_generic.go

```diff
@@ -57,6 +82,323 @@ func NewUartGeneric(dev string, baud int, timeout int) *UartGeneric {
+                        return nil, err
+                }
+        case common.ValueTypeFloat64:
+                val, err = cast.ToFloat64E(reading)
+                if err != nil {
+                        return nil, fmt.Errorf(castError, resource.Name, err)
+                }
+                if err := checkFloatValueRange(valueType, val); err != nil {
+                        return nil, err
+                }
+        default:
+                return nil, fmt.Errorf("return result fail, unsupported value type: %v", valueType)
+        }
+
+        return val, nil
+}
+
+func checkUintValueRange(valueType string, val interface{}) error {
+        switch valueType {
+        case common.ValueTypeUint8:
+                _, ok := val.(uint8)
+                if ok {
+                        return nil
+                }
+        case common.ValueTypeUint16:
+                _, ok := val.(uint16)
+                if ok {
+                        return nil
+                }
+        case common.ValueTypeUint32:
+                _, ok := val.(uint32)
+                if ok {
+                        return nil
+                }
+        case common.ValueTypeUint64:
+                _, ok := val.(uint64)
+                if ok {
+                        return nil
+                }
+        }
+        return fmt.Errorf("value %v for %s type is out of range", val, valueType)
+}
+
+func checkIntValueRange(valueType string, val interface{}) error {
+        switch valueType {
+        case common.ValueTypeInt8:
+                _, ok := val.(int8)
+                if ok {
+                        return nil
+                }
+        case common.ValueTypeInt16:
+                _, ok := val.(int16)
+                if ok {
+                        return nil
+                }
+        case common.ValueTypeInt32:
+                _, ok := val.(int32)
+                if ok {
+                        return nil
+                }
+        case common.ValueTypeInt64:
+                _, ok := val.(int64)
+                if ok {
+                        return nil
+                }
+        }
+        return fmt.Errorf("value %v for %s type is out of range", val, valueType)
+}
+
+func checkFloatValueRange(valueType string, val interface{}) error {
+        switch valueType {
+        case common.ValueTypeFloat32:
+                valFloat := val.(float32)
+                if math.Abs(float64(valFloat)) >= math.SmallestNonzeroFloat32 && math.Abs(float64(valFloat)) <= math.MaxFloat32 {
+                        return nil
+                }
+        case common.ValueTypeFloat64:
+                valFloat := val.(float64)
+                if math.Abs(valFloat) >= math.SmallestNonzeroFloat64 && math.Abs(valFloat) <= math.MaxFloat64 {
+                        return nil
+                }
+        }
+
+        return fmt.Errorf("value %v for %s type is out of range", val, valueType)
+}
+
```

Comment:
Why is all this new code being added?

This is out of scope for migrating to 3.0, please remove.

Reply:
Removed. 

We were having plan to support multiple datatypes i.e., Read and write integers. Hence, I had a thought of having it here. However, removed it as it is not in v3 migration scope.